src/index.js

```diff
@@ -84,7 +85,11 @@ async function republishPackage(originPackageIdentifier, target, publishArgs = [
 
     subProcess.stderr.pipe(process.stderr)
     subProcess.stdout.pipe(process.stdout)
-  }).then(cleanUp)
+  }).then(cleanUp).then(() => {
+    if (shouldUnpublish) {
+      unpublishPackage(registry.to, originPackageName, originPackageVersion);
```

Comment:
in case unpublishPackage will be async one day, let's return the returned value so we won't have a bug if we forget to do it later. 

I fixed similar bug in wix-fed-scripts not a while ago.

Reply:
Good catch!